shared/teams/team/member/add-to-channels.tsx

```diff
@@ -186,96 +189,224 @@ const AddToChannels = (props: Props) => {
         <Kb.Box2 direction="vertical" style={Styles.globalStyles.flexOne}>
-          <Kb.Box2 direction="horizontal" gap="small" alignSelf="flex-start">
+          <Kb.Box2 direction="horizontal" gap="tiny" alignSelf="flex-start">
             <Kb.Text type="Body" lineClamp={1} style={styles.channelText}>
-              #{channelname}
+              #{channelMeta.channelname}
             </Kb.Text>
             <Common.ParticipantMeta numParticipants={numParticipants} />
           </Kb.Box2>
           <Common.Activity level={activityLevel} />
         </Kb.Box2>
-        <Kb.CheckCircle checked={selected} onCheck={onSelect} disabled={channelname === 'general'} />
+        {selfMode ? (
+          <SelfChannelActions meta={channelMeta} />
+        ) : (
+          <Kb.CheckCircle
+            checked={selected}
+            onCheck={onSelect}
+            disabled={channelMeta.channelname === 'general'}
+          />
+        )}
       </Kb.Box2>
     </Kb.ClickableBox>
   ) : (
     <Kb.ListItem2
-      onMouseDown={evt => {
-        // using onMouseDown so we can prevent blurring the search filter
-        evt.preventDefault()
-        onSelect()
-      }}
+      onMouseDown={
+        selfMode
+          ? undefined
+          : evt => {
+              // using onMouseDown so we can prevent blurring the search filter
+              evt.preventDefault()
+              onSelect()
+            }
+      }
       type="Large"
       action={
-        <Kb.CheckCircle
-          checked={selected}
-          onCheck={() => {
-            /* ListItem onMouseDown triggers this */
-          }}
-        />
+        selfMode ? (
+          <SelfChannelActions meta={channelMeta} />
+        ) : (
+          <Kb.CheckCircle
+            checked={selected}
+            onCheck={() => {
+              /* ListItem onMouseDown triggers this */
+            }}
+          />
+        )
       }
+      height={selfMode ? 72 : undefined}
       firstItem={true}
       body={
         <Kb.Box2 direction="vertical" alignItems="stretch">
-          <Kb.Text type="BodySemibold" lineClamp={1}>
-            #{channelname}
-          </Kb.Text>
-          <Kb.Box2 direction="horizontal" alignSelf="stretch" gap="xxtiny">
-            <Kb.Text type="BodySmall">
-              {numMembers} {pluralize('member', numMembers)} •
+          <Kb.Box2 direction="horizontal" gap="xtiny" alignSelf="flex-start">
+            <Kb.Text type="Body" lineClamp={1} style={styles.channelText}>
+              #{channelMeta.channelname}
             </Kb.Text>
-            <Common.Activity level="recently" />
+            {selfMode && <Common.ParticipantMeta numParticipants={numParticipants} />}
           </Kb.Box2>
+          <Kb.Box2 direction="horizontal" alignSelf="stretch" gap="xxtiny">
+            {!selfMode && (
+              <Kb.Text type="BodySmall">
+                {numMembers} {pluralize('member', numMembers)} •
```

Comment:
one of `numMembers` and `numParticipants` should go. `numParticipants` is the good one in my testing.